int32[1] data
